File: content-solutions/queue-deque/deque-end-cards.js
function solution(cards) {
  const score = [0, 0];
  let left = 0;
  let right = cards.length - 1;
  let turn = 0;
  while (left <= right) {
    const card = cards[left] >= cards[right] ? cards[left++] : cards[right--];
    score[turn % 2] += card;
    turn += 1;
  }
  return score;
}

File: content-solutions/queue-deque/deque-end-cards.java
class Solution {
    public int[] solution(int[] cards) {
        int[] score = new int[2];
        int left = 0, right = cards.length - 1, turn = 0;
        while (left <= right) {
            int card = cards[left] >= cards[right] ? cards[left++] : cards[right--];
            score[turn % 2] += card;
            turn++;
        }
        return score;
    }
}
